No debug info when building Kubernetes source code with `make all GOGCFLAGS="-N -l"`

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:
I ran `make all GOGCFLAGS="-N -l"` to build latest Kubernetes source code, and then I see the binaries are still stripped:
```
$ file _output/bin/kubelet 
_output/bin/kubelet: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=715f57807e652488c5e086b82f5daf4063a74b04, stripped
```

And I can not use dlv to debug the binary:
```
$ dlv exec _output/bin/kubelet                          
could not launch process: could not open debug info
```

In Makefile, I see the following comments:
```
#   make all GOGCFLAGS="-N -l"
#     Note: Use the -N -l options to disable compiler optimizations an inlining.
#           Using these build options allows you to subsequently use source
#           debugging tools like delve.
```
But that seems not correct in my build.

**What you expected to happen**:
I expected Kubernetes binaries with debug info.

**How to reproduce it (as minimally and precisely as possible)**:
`make all GOGCFLAGS="-N -l"`

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): Latest Kubernetes source code.
- Cloud provider or hardware configuration: A virtual machine with 2 cores and 8GB RAM.
- OS (e.g: `cat /etc/os-release`): 18.04 LTS (Bionic Beaver)
- Kernel (e.g. `uname -a`): Linux workstation 4.15.0-23-generic #25-Ubuntu SMP Wed May 23 18:02:16 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

https://github.com/kubernetes/kubernetes/blob/master/hack/lib/golang.sh

kube::golang::build_binaries() {
  # Create a sub-shell so that we don't pollute the outer environment
  (
    # Check for `go` binary and set ${GOPATH}.
    kube::golang::setup_env
    V=2 kube::log::info "Go version: $(go version)"

    local host_platform
    host_platform=$(kube::golang::host_platform)

    local goflags goldflags goasmflags **gogcflags**
    # If GOLDFLAGS is unset, then set it to the a default of "-s -w".
    # Disable SC2153 for this, as it will throw a warning that the local
    # variable goldflags will exist, and it suggest changing it to this.
    # shellcheck disable=SC2153
    goldflags="${GOLDFLAGS=-s -w} $(kube::version::ldflags)"
    goasmflags="-trimpath=${KUBE_ROOT}"
    gogcflags="${GOGCFLAGS:-} -trimpath=${KUBE_ROOT}"    
           ~~~~~ It is a local variable 
           ~~~~~ It turns out the value is empty in kube::golang::build_binaries_for_platform


I will take this one and provide a PR.

/cc @YoubingLi

--- a/build/root/Makefile
+++ b/build/root/Makefile
@@ -78,10 +78,11 @@ define ALL_HELP_INFO
 #   make
 #   make all
 #   make all WHAT=cmd/kubelet GOFLAGS=-v
-#   make all GOGCFLAGS="-N -l"
-#     Note: Use the -N -l options to disable compiler optimizations an inlining.
-#           Using these build options allows you to subsequently use source
-#           debugging tools like delve.
+#   make all GOLDFLAGS=""
+#     Note: Specify GOLDFLAGS as an empty string for building unstripped binaries, which allows
+#           you to use code debugging tools like delve. When GOLDFLAGS is unspecified, it defaults
+#           to "-s -w" which strips debug information. Other flags that can be used for GOLDFLAGS 
+#           are documented at https://golang.org/cmd/link/
 endef
 .PHONY: all
 ifeq ($(PRINT_HELP),y)

